### ISSUE TITLE
Package version conflict

### ISSUE THREAD

**Issue by bigssmart** [instadeepai/og-marl#18]
Thank you very much for your outstanding contributions in this field. I have some questions about installing related packages.

tensorstore 0.1.54 requires ml-dtypes>=0.3.1.

But tensorflow-intel 2.15.0 requires ml-dtypes~=0.2.0.

And flashbax 0.1.0 requires tensorstore>=0.1.51

Could you please provide more details about the version of the related package? Thank you very much

**Comment by jcformanek** [comment]
Thanks so much for opening an issue. It looks like you solved the problem, but I thought I would leave a comment anyway. 

We are aware of this dependency issue. But as long as you only install Flashbax **after** installing OG-MARL and its other dependencies, then everything will work. 

If you have any other issues using OG-MARL, please do not hesitate to open another issue. Or if you just have a question, we would be happy to answer. 

Good luck with your research!

**Comment by bigssmart** [comment]
> 非常感谢您提出问题。看来你解决了问题，但我想我还是要发表评论。
> 
> 我们意识到这个依赖性问题。但只要您在安装 OG-MARL 及其其他依赖项**之后才安装 Flashbax，那么一切都会正常。**
> 
> 如果您在使用 OG-MARL 时遇到任何其他问题，请随时打开另一个问题。或者，如果您只是有一个问题，我们很乐意回答。
> 
> 祝您研究顺利！

**Comment by bigssmart** [comment]
Thank you very much for your reply. I reproduced qmix+cql in 3m_good scenario and found always starts to rapidly decline in performance after fewer than 1000 training steps。At present, the parameters in the code are used. Does this part still need additional parameters? It would be very helpful if you could provide me with more details. Thanks again for your patience and reply!

num_ood_actions: int = 10,
        cql_weight: float = 5.0,
        linear_layer_dim: int = 64,
        recurrent_layer_dim: int = 64,
        mixer_embed_dim: int = 32,
        mixer_hyper_dim: int = 64,
        discount: float = 0.99,
        target_update_period: int = 200,
        learning_rate: float = 3e-4,

![X42(~YD3%AH $Y`)TND
BF](https://github.com/instadeepai/og-marl/assets/61147677/90b6c804-2206-423a-a99d-ccb8a37e5d32)

Or could it be the result of the following warnings?
WARNING:tensorflow:From D:\ANACONDA\anaconda\envs\ogmarl\lib\site-packages\keras\src\losses.py:2976:  The name tf.losses.sparse_softmax_cross_entropy is deprecated. Please use  tf.compat.v1.losses.sparse_softmax_cross_entropy instead.

D:\ANACONDA\anaconda\envs\ogmarl\lib\site-packages\flashbax\buffers\trajectory_buffer.py:473: UserWarning:  Setting max_size dynamically sets the `max_length_time_axis` to be `max_size`//`add_batch_size = 50000`.This allows one  to control exactly how many timesteps are stored in the buffer.Note that this overrides the `max_length_time_axis`  argument.
warnings.warn(

D:\ANACONDA\anaconda\envs\ogmarl\lib\site-packages\flashbax\buffers\trajectory_buffer.py:498: UserWarning:  `sample_sequence_length` greater than `min_length_time_axis`,  therefore overriding `min_length_time_axis`to be set to `sample_sequence_length`,  as we need at least `sample_sequence_length` timesteps added to the buffer before we can sample.
warnings.warn(

**Comment by jcformanek** [comment]
Hi there, here are my latest baseline results: https://api.wandb.ai/links/off-the-grid-marl-team/f2ozf5vy

I do find that the performance does decay after extended training. But 1000 training steps does seem too soon. In my experiments performance decayed after 30k training steps. Could you double check that you are not running Qmix without CQL by mistake?

**Comment by jcformanek** [comment]
Oh wait, I see your x-axis there is "Step", which is logging steps not training steps. You should change the x-axis on the `evaluator/episode_return` to `Trainer Steps (eval)`. Then you will see the training steps on the x-axis.

But yes, its expected that after long training the performance begins to degrade. This is because over time the algorithm trains on more and more out-of-distribution actions. This results in compounding extrapolation and eventually performance collapse. Other algorithms are more robust to this and do no suffer from performance degradation so soon (e.g. MAICQ and IDRQN+CQL)

**Comment by bigssmart** [comment]
Thank you very much for your reply!
